src/Compilers/Shared/RuntimeHostInfo.cs

```diff
@@ -39,17 +38,17 @@ internal static (string processFilePath, string commandLineArguments, string too
             var dotnetFilePath = GetDotNetPathOrDefault();
             commandLineArguments = $@"exec ""{toolFilePath}"" {commandLineArguments}";
             return (dotnetFilePath, commandLineArguments, toolFilePath);
-#else
-            var toolFilePath = $"{toolFilePathWithoutExtension}.exe";
-            return (toolFilePath, commandLineArguments, toolFilePath);
-#endif
         }
 
+        internal static bool IsCoreClrRuntime =>
 #if NET
-
-        internal static bool IsCoreClrRuntime => true;
+            true;
+#else
+            false;
+#endif
 
         private const string DotNetHostPathEnvironmentName = "DOTNET_HOST_PATH";
+        private const string DotNetExperimentalHostPathEnvironmentName = "DOTNET_EXPERIMENTAL_HOST_PATH";
```

Comment:
I didn't understand why we need this now.

Reply:
The plan was that full-framework MSBuild.exe will set DOTNET_HOST_PATH so build tasks can use it, but for some reason that was (temporarily?) changed to DOTNET_EXPERIMENTAL_HOST_PATH in https://github.com/dotnet/sdk/pull/46248.